doc/neps/nep-0028-website-redesign.rst

```diff
@@ -0,0 +1,302 @@
+now contains only a couple of pages and may contain on the order of ten
+pages after the redesign. Changing the documentation (user guide, reference
+guide, and some other pages in the NumPy Manual) is out of scope for
+this proposal.
+
+
+Detailed description
+--------------------
+
+User Experience
+~~~~~~~~~~~~~~~
+
+Besides the NumPy logo, there is little that can or needs to be kept from the
+current website. We will rely to a large extent on ideas and proposals by the
+designer(s) of the new website.
+
+As reference points we can use the `Jupyter website <https://jupyter.org/>`_,
+which is probably the best designed site in our ecosystem, and the
+`QuantEcon <https://quantecon.org>`_ and `Julia <https://julialang.org>`_
+sites which are well-designed too.
+
+The Website
+~~~~~~~~~~~
+
+A static site is a must. There are many high-quality static site generators.
+The current website uses Sphinx, however that is not the best choice - it's
+hard to theme and results in sites that are too text-heavy due to Sphinx'
+primary aim being documentation.
+
+The following should be considered when choosing a static site generator:
+
+1. *How widely used is it?* This is important when looking for help maintaining
+   or improving the site. More popular frameworks are usually also better
+   maintained, so less chance of bugs or obsolescence.
+2. *Ease of deployment.* Most generators meet this criterion, however things
+   like built-in support for GitHub Pages helps.
+3. *Preferences of who implements the new site.* Everyone has their own
+   preferences. And it's a significant amount of work to build a new site.
+   So we should take the opinion of those doing the work into account.
+
+Traffic
+```````
+
+The current site receives on the order of 500,000 unique visitors per month.
+With a redesigned site and relevant content, there is potential for visitor
+counts to reach 5-6 million -- a similar level as
+`scipy.org <http://scipy.org>`_ or `matplotlib.org <http://matplotlib.org>`_ --
+or more.
+
+Possible options for static site generators
+```````````````````````````````````````````
+
+1. *Jekyll.* This is a well maintained option with 855 Github contributors,
+   with contributions within the last month. Jekyll is written in Ruby, and
+   has a simple CLI interface. Jekyll also has a large directory of
+   `themes <https://jekyllthemes.io>`__, although a majority cost money.
+   There are several themes (`serif <https://jekyllthemes.io/theme/serif>`_,
+   `uBuild <https://jekyllthemes.io/theme/ubuild-jekyll-theme>`_,
+   `Just The Docs <https://jekyllthemes.io/theme/just-the-docs>`_) that are
+   appropriate and free. Most themes are likely responsive for mobile, and
+   that should be a requirement. Jekyll uses a combination of liquid templating
+   and YAML to render HTML, and content is written in Markdown. i18n
+   functionality is not native to Jekyll, but can be added easily.
```

Comment:
It's possibly worth noting that unlike the others, Jekyll's site-generator is run automatically by github. Perhaps this is a non-issue and we can set up CI to do this step.

Reply:
Indeed. It's a small thing, but nice to have. CI does break once in a while.

Reply:
> It's possibly worth noting that unlike the others, Jekyll's site-generator is run automatically by github pages

If GitLab could be an option for hosting the website, then it is much better than GitHub because it allows you to use any site-generator that you wish. This is so because it it allows you to run scripts and commands for generating the site yourself. There are also examples for different site-generators (including sphinx): https://gitlab.com/pages

Reply:
Not that it's my business, but having everything on one platform is very much the power of the ecosystem. Moving some elements to gitlab, I feel would be counter productive on the medium-long run.

Reply:
I'm unsure if Gitlab offers anything Github doesn't. Both Gitlab and Github should be completely agnostic to which generator we end up with. Github has Github Integrations, which work quite well for deployment/CI.

Reply:
added the GitHub Pages hook as an additional benefit